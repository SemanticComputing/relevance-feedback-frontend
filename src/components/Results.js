import React from 'react';
import { object, func } from 'prop-types';
import { map } from 'lodash';
import Result from '../components/Result';

const Results = ({ results, updateThumb, describeTopic }) => {
  return map(results.items, (result) => (
    <Result key={result.url} result={result} updateThumb={updateThumb}
      describeTopic={describeTopic} currentTopic={results.currentTopic} />
  ));
};

Results.propTypes = {
  results: object,
  updateThumb: func,
  describeTopic: func
};

export default Results;
