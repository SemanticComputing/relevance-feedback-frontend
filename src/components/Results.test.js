import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Results from './Results';

const updateThumb = jest.fn();
const results = [
  {
    name: 'title',
    url: 'http://example.fi',
    description: 'description'
  },
  {
    name: 'other title',
    url: 'http://another.fi',
    description: 'other description'
  },
];

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Results updateThumb={updateThumb} result={results} />, div);
});

it('renders results', () => {
  const tree = renderer
    .create(<Results updateThumb={updateThumb} results={results} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
