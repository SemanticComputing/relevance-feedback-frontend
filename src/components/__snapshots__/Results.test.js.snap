// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders results 1`] = `
Array [
  <div
    className="result row"
>
    <div
        className="col-sm-8 col"
    >
        <div
            className="row"
        >
            <div
                className="col"
            >
                <a
                    href="http://example.fi"
                >
                    title
                </a>
            </div>
        </div>
        <div
            className="row"
        >
            <div
                className="col"
            >
                description
            </div>
        </div>
    </div>
    <div
        className="col-sm-2 col"
    >
        <div
            className="row"
        >
            <div
                className="col"
            >
                <button
                    className="btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                >
                    <span
                        aria-hidden={true}
                        className="fa fa-thumbs-up fa-2x thumbs thumbs-up"
                        style={Object {}}
                    />
                </button>
            </div>
            <div
                className="col"
            >
                <button
                    className="btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                >
                    <span
                        aria-hidden={true}
                        className="fa fa-thumbs-down fa-2x thumbs thumbs-down"
                        style={Object {}}
                    />
                </button>
            </div>
        </div>
    </div>
</div>,
  <div
    className="result row"
>
    <div
        className="col-sm-8 col"
    >
        <div
            className="row"
        >
            <div
                className="col"
            >
                <a
                    href="http://another.fi"
                >
                    other title
                </a>
            </div>
        </div>
        <div
            className="row"
        >
            <div
                className="col"
            >
                other description
            </div>
        </div>
    </div>
    <div
        className="col-sm-2 col"
    >
        <div
            className="row"
        >
            <div
                className="col"
            >
                <button
                    className="btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                >
                    <span
                        aria-hidden={true}
                        className="fa fa-thumbs-up fa-2x thumbs thumbs-up"
                        style={Object {}}
                    />
                </button>
            </div>
            <div
                className="col"
            >
                <button
                    className="btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                >
                    <span
                        aria-hidden={true}
                        className="fa fa-thumbs-down fa-2x thumbs thumbs-down"
                        style={Object {}}
                    />
                </button>
            </div>
        </div>
    </div>
</div>,
]
`;
